internal/connection/connection.go

```diff
@@ -102,6 +224,12 @@ func (c *Connection) fetchCapabilities(ctx context.Context) error {
 	return nil
 }
 
+// ensureConnected re-eastablishes a connection if it is in a disconnected
+// state. An error is returned if it fails to renew the connection.
+func (c *Connection) ensureConnected() error {
+	return c.Connect()
+}
```

Comment:
Directly call `c.Connect()` ?

Reply:
Let's keep `ensureConnected`, for readability purposes? It is _a little_ more verbose about the intent in places outside of `connection.go`.